servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/TrailersOnlyErrorTest.java

```diff
@@ -0,0 +1,254 @@
+
+        final ServerContext serverContext = GrpcServers.forAddress(localAddress(0))
+                .listenAndAwait(new TesterProto.Tester.ServiceFactory(service));
+
+        final GrpcClientBuilder<HostAndPort, InetSocketAddress> clientBuilder =
+                GrpcClients.forAddress(serverHostAndPort(serverContext))
+                        .appendHttpClientFilter(__ -> true, setupResponseVerifierFilter(asyncErrors));
+
+        TesterProto.Tester.TesterClient client = clientBuilder.build(new TesterProto.Tester.ClientFactory());
+        verifyException(client.test(TesterProto.TestRequest.newBuilder()
+                .build()).toFuture(), UNKNOWN);
+        assertNoErrors(asyncErrors);
+    }
+
+    @Test
+    public void testServiceFilterThrows() throws Exception {
+        final BlockingQueue<Throwable> asyncErrors = new LinkedBlockingDeque<>();
+        final TesterProto.Tester.TesterService service = mockTesterService();
+
+        final ServerContext serverContext = GrpcServers.forAddress(localAddress(0))
+                .appendHttpServiceFilter(svc -> new StreamingHttpServiceFilter(svc) {
+                    @Override
+                    public Single<StreamingHttpResponse> handle(
+                            final HttpServiceContext ctx, final StreamingHttpRequest request,
+                            final StreamingHttpResponseFactory responseFactory) {
+                        throw DELIBERATE_EXCEPTION;
+                    }
+                })
+                .listenAndAwait(new TesterProto.Tester.ServiceFactory(service));
+
+        final GrpcClientBuilder<HostAndPort, InetSocketAddress> clientBuilder =
+                GrpcClients.forAddress(serverHostAndPort(serverContext))
+                        .appendHttpClientFilter(__ -> true, setupResponseVerifierFilter(asyncErrors));
+
+        TesterProto.Tester.TesterClient client = clientBuilder.build(new TesterProto.Tester.ClientFactory());
+        verifyException(client.test(TesterProto.TestRequest.newBuilder()
+                .build()).toFuture(), UNKNOWN);
+        assertNoErrors(asyncErrors);
+
+        verifyException(client.testRequestStream(Publisher.from(TesterProto.TestRequest.newBuilder()
+                .build())).toFuture(), UNKNOWN);
+        assertNoErrors(asyncErrors);
+
+        verifyException(client.testResponseStream(TesterProto.TestRequest.newBuilder().build()).toFuture(), UNKNOWN);
+        assertNoErrors(asyncErrors);
+
+        verifyException(client.testBiDiStream(from(TesterProto.TestRequest.newBuilder()
+                .build()).concat(never())).toFuture(), UNKNOWN);
+        assertNoErrors(asyncErrors);
+
+        verifyException(client.testBiDiStream(from(TesterProto.TestRequest.newBuilder()
+                .build())).toFuture(), UNKNOWN);
+        assertNoErrors(asyncErrors);
+    }
+
+    private static void assertNoErrors(final BlockingQueue<Throwable> errors) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
We already have a utility like this in `H2PriorKnowledgeFeatureParityTest` and `AsyncContextHttpFilterVerifier`. `servicetalk-test-resources` might be a good place to share it.